net_setup.go

```diff
@@ -0,0 +1,118 @@
+package main
+
+import (
+	"fmt"
+	"net"
+)
+
+var (
+	wireguardConfig struct {
+		gatewayIPv4 net.IP
+		gatewayIPv6 net.IP
+		networkIPv4 *net.IPNet
+		networkIPv6 *net.IPNet
+	}
+)
+
+func initWireguardConfig() error {
+	var err error
+	wireguardConfig.gatewayIPv4, wireguardConfig.networkIPv4, err = calcDefaultGateway(networkIPv4)
+	if err != nil {
+		return err
+	}
+	wireguardConfig.gatewayIPv6, wireguardConfig.networkIPv6, err = calcDefaultGateway(networkIPv6)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func configureWireguard() (string, error) {
+
+	type value struct {
+		Nameserver string
+		NetworkIPv4 string
+		IPv6NatEnabled bool
+		NetworkIPv6 string
+		GatewayIPv4 string
+		GatewayIPv6 string
+		GatewayIPv4WithCIDR string
+		GatewayIPv6WithCIDR string
+		DnsmasqEnabled bool
+	}
+	maskLenIPv4, _ := wireguardConfig.networkIPv4.Mask.Size()
+	maskLenIPv6, _ := wireguardConfig.networkIPv6.Mask.Size()
+	v := value{
+		Nameserver:          nameserver,
+		NetworkIPv4:         networkIPv4,
+		IPv6NatEnabled:      ipv6NatEnabled,
+		NetworkIPv6:         networkIPv6,
+		GatewayIPv4:         wireguardConfig.gatewayIPv4.String(),
+		GatewayIPv6:         wireguardConfig.gatewayIPv6.String(),
+		GatewayIPv4WithCIDR: fmt.Sprintf("%s/%d", wireguardConfig.gatewayIPv4.String(), maskLenIPv4),
+		GatewayIPv6WithCIDR: fmt.Sprintf("%s/%d", wireguardConfig.gatewayIPv6.String(), maskLenIPv6),
+		DnsmasqEnabled: enableDnsmasq,
+	}
+	return bash(`
```

Comment:
i know this would extend the amount of work you have already put in, prehaps work for a future PR. Would it be possible to implement `go-iptables` by coreos for interactions with `ipTables` ?

Reply:
OK, I will try using this library:
https://github.com/coreos/go-iptables